### PR TITLE
Regression tests

### PR DESCRIPTION
**Pull request by blue-moon22**
Here is an attempt at testing outputs of nextflow in test.sh.

It runs the pipeline on a 'test' sample and compares the output with the references. It takes about 2 minutes on my machine. It would be better to test on multiple samples but it would be difficult to do locally and also without using some kind of large file storage. Maybe if this is possible with Github LFS or similar we could implement this in Github CI?

Another option is to use smaller simulated data but I'm slightly wary of this because creating representative simulated data is another challenge in itself.

Any suggesions welcome. Happy to keep this discussion open and have this as a separate branch for now.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
```
Comment (kpepper) [inline test.sh:2]: Add a "Starting regression tests..." echo

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    diff_status=$(diff -s ${test} ${reference})
+    # If files different then warning
+    if [[ $diff_status != "Files ${test} and ${reference} are identical" ]]; then
+        echo "${test} is missing or its contents not expected."
+        return 1
+    fi
+}
+
+error_status=0
+# Check for test_drug_cat_alleles.txt output
```
Comment (kpepper) [inline test.sh:22]: Add quotes around the file path args for the file_diff calls.

**Review by kpepper — APPROVED** [review]
This is something we can build on as required, so definitely worth adding in. If you have time add it into the GitHub build workflow (will need to install nextflow in the workflow setup).

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+# Run pipeline on test input data
+nextflow run main.nf --reads "$in_dir/test_{1,2}.fastq.gz" --results_dir "$out_dir" --output 'test'
+
+function file_diff {
+    local test=${1}
+    local reference=${2}
+
+    diff_status=$(diff -s ${test} ${reference})
+    # If files different then warning
+    if [[ $diff_status != "Files ${test} and ${reference} are identical" ]]; then
```
[the remark this answers is not in the document]
Comment (blue-moon22) [reply test.sh:15]: Not sure what you mean. How do I then compare the output of diff to certain cases?

[the remark this answers is not in the document]
Comment (kpepper) [reply test.sh:15]: diff ${test} ${reference} > /dev/null 2>&1
if [[ $? -gt 0 ]]
then
   echo ...
   return 1
fi
- Just avoids having to rely on the output text of diff.

**Review by kpepper — APPROVED** [review]
Looks good!

**Review by kpepper — APPROVED** [review]

**Pull request merged**